test/domain/AuthTokenTest.js

```diff
@@ -43,16 +43,6 @@ describe("AuthToken", function () {
       expect(authToken.tokenIdIsValid()).is.false;
       expect(authToken.isValid()).is.false;
 
-      // Invalid field value
-      authToken.tokenId = new Date();
-      expect(authToken.tokenIdIsValid()).is.false;
-      expect(authToken.isValid()).is.false;
```

Comment:
If you use Javascript's native BigInt function and call `BigInt(date)`, it will convert the date to a timestamp, represented as a valid number

![Screen Shot 2023-06-27 at 14 19 50](https://github.com/bosonprotocol/boson-protocol-contracts/assets/50335980/04bf3adc-d42f-4777-bbda-7bbefca115ab)

Reply:
I see. But the conversion happens only in `bigNumberIsValid`, while the actual stored value is still a date (we don't convert it in the constructor).
So although the field looks it's valid, it might not have all the expected properties.

In the domain tests, we state for this type of fields that "must be the *string* representation of a BigNumber".
So I guess it would make sense to add `typeof bigNumber === "string"` in `bigNumberIsValid` and then date won't be valid anymore.

There is one thing that would be generally even better - we could check `typeof bigNumber === "bigint"` so we won't use string representation anymore. But that would probably affect a lot of tests, so it's probably not worth doing it now.

Reply:
fixed and tests added back